gcloud-java-dns/src/main/java/com/google/gcloud/dns/DnsRecord.java

```diff
@@ -0,0 +1,280 @@
+/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.gcloud.dns;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.common.base.MoreObjects;
+import com.google.common.collect.ImmutableList;
+
+import java.io.Serializable;
+
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * A class that represents Google Cloud DNS record set.
+ *
+ * <p>A unit of data that will be returned by the DNS servers.
+ *
+ * @see <a href="https://cloud.google.com/dns/api/v1/resourceRecordSets">Google Cloud DNS
+ * documentation</a>
+ */
+public class DnsRecord implements Serializable {
+
+  private static final long serialVersionUID = 2016011914302204L;
+  private final String name;
+  private final List<String> rrdatas;
+  private final Integer ttl;
+  private final DnsRecordType type;
+  private final String zoneName;
+  private final Long zoneId;
+
+  private DnsRecord() {
+    this.name = null;
+    this.rrdatas = null;
+    this.ttl = null;
+    this.type = null;
+    this.zoneName = null;
+    this.zoneId = null;
+  }
+
+  DnsRecord(Builder builder) {
+    this.name = builder.name;
+    this.rrdatas = ImmutableList.copyOf(builder.rrdatas);
+    this.ttl = builder.ttl;
+    this.type = builder.type;
+    this.zoneName = builder.zoneName;
+    this.zoneId = builder.zoneId;
+  }
+
+  /**
+   * Enum for the DNS record types supported by Cloud DNS.
+   *
+   * <p>Google Cloud DNS currently supports records of type A, AAAA, CNAME, MX NAPTR, NS, PTR, SOA,
+   * SPF, SRV, TXT.
+   *
+   * @see <a href="https://cloud.google.com/dns/what-is-cloud-dns#supported_record_types">Cloud DNS
+   * supported record types</a>
+   */
+  public enum DnsRecordType {
+    A,
```

Comment:
Would be nice to add item level javadoc from [here](https://cloud.google.com/dns/what-is-cloud-dns).
e.g. "Address record, which is used to map host names to their IPv4 address." for `A`